external-import/cve/src/services/client/api.py

```diff
@@ -64,10 +64,20 @@ def request(self, api_url, params):
             # It is recommended that users "sleep" their scripts for six seconds between requests (NIST)
             time.sleep(6)
             return response
-        else:
-            raise Exception(
-                "[API] Attempting to retrieve data failed. Wait for connector to re-run..."
-            )
+        elif response.status_code == 404:
+            error_data = response.headers
+            if error_data.get("message") == "Invalid apiKey.":
+                error_msg = (
+                    "[API] Invalid API Key provided. Please check your configuration."
+                )
+                self.helper.connector_logger.error(error_msg, meta={"error": error_msg})
+                raise Exception(error_msg)
+            else:
+                self.helper.connector_logger.error(f"[API] Error: {error_data.get('message')}",
+                                                  meta={"error": error_data.get("message")})
```

Comment:
```suggestion
        if response.status_code == 200:
            # It is recommended that users "sleep" their scripts for six seconds between requests (NIST)
            time.sleep(6)
            return response
        elif response.status_code == 404:
            error_data = response.headers
            if error_data.get("message") == "Invalid apiKey.":
                raise Exception(
                    "[API] Invalid API Key provided. Please check your configuration."
                )
            else:
                raise Exception(f"[API] Error: {error_data.get('message')}")
```
As the error is catched and logged in `_request_data` without re raising, you log 2 times the same messages